content/blogposts/2022/release-3.42

```diff
@@ -0,0 +1,55 @@
+---
+title: "Sourcegraph 3.42 release"
+publishDate: 2022-07-22T10:00-07:00
+description: "Sourcegraph 3.42 introduces UPDATE"
+tags: [blog, release]
+slug: "release/3.42"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+
+---
+
+Sourcegraph 3.42 is now available! For this release, we introduced:
+
+## All-new Admin Analytics 
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, and ultimately better manage user licenses, identify power users, and convey value to their internal leaders. 
+
+## Improved search-based navigation for Python non-locals
+We previously announced we’d [improved our search-based support for Python locals in Sourcegraph 3.40(https://about.sourcegraph.com/blog/release/3.40) by using tree-sitter. We've now extended this improvement for jump-to-definition to more kinds of variables such as class fields.
+
+## Beta UI improvements
+We're fixing UX papercuts and cleaning up our UI. 3.42 will have an optional Beta UI, which can be enabled via a toggle in the avatar menu. It's disabled by default but we encourage you to give it a spin! We think you'll find it cleaner and easier to use. You can find the toggle just below the theme dropdown in the avatar menu.
+
+## Searching by repository descriptions
+We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## New search suggestions
```

Comment:
@benvenker What's the customer impact/benefit of this? Will it feel significantly different to the average user?

Reply:
@iskyOS it means users can search repositories by topic (as long as the repo description has topical keywords in it). It should feel significantly different to anyone who wants to find "types" of repos as opposed to a specific repo.

Reply:
@benvenker, asking about CodeMirror implementation / new search suggestions

Reply:
Whoops. For now, it means small improvements in speed and content in suggestions. Long term it will enable us to dramatically improve suggestion content.